src/fare-contracts/FareContractAndReservationsList.tsx

```diff
@@ -33,12 +35,28 @@ export const FareContractAndReservationsList: React.FC<Props> = ({
 }) => {
   const navigation = useNavigation<RootNavigationProp>();
   const analytics = useAnalytics();
+  const {abtCustomerId: currentUserId} = useAuthState();
+
+  const calculateWeight = useCallback(
+    (fcOrReservation: FareContract | Reservation) => {
+      const isFareContract = 'travelRights' in fcOrReservation;
+      // Make reservations go first, then fare contracts
+      if (!isFareContract) return 1;
+
+      const fc = getFareContractInfo(now, fcOrReservation, currentUserId);
+      return fc.validityStatus === 'valid' ? 1 : 0;
+    },
+    [now, currentUserId],
+  );
 
   const fareContractsAndReservationsSorted = useMemo(() => {
-    return [...(fareContracts || []), ...(reservations || [])].sort(
-      (a, b) => b.created.toMillis() - a.created.toMillis(),
-    );
-  }, [reservations, fareContracts]);
+    return [...(fareContracts || []), ...(reservations || [])].sort((a, b) => {
+      const validityWeight = calculateWeight(b) - calculateWeight(a);
```

Comment:
Could `validityWeight` be called `weightDelta` instead? 
Suggesting this because `calculateWeight` considers more than just validity, and since it is a difference between the weights, which means it can also be -1, not just 1 or 0.